SimplenoteWidgets/Tools/WidgetDataController.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import SimplenoteFoundation
+import SimplenoteSearch
+
+class WidgetDataController {
+
+    /// Data Controller
+    ///
+    let coreDataManager: CoreDataManager
+
+    /// Notes Controller
+    ///
+    private lazy var notesController = ResultsController<Note>(
+        viewContext: coreDataManager.managedObjectContext,
+        matching: predicateForNotes(),
+        sortedBy: [NSSortDescriptor.descriptorForNotes(sortMode: noteSortMode)]
+    )
+
+    /// Initialization
+    ///
+    init(coreDataManager: CoreDataManager) throws {
+        guard let isLoggedIn = UserDefaults(suiteName: SimplenoteConstants.sharedGroupDomain)?.bool(forKey: .accountIsLoggedIn),
+              isLoggedIn else {
+            throw StorageError.appConfigurationError
+        }
+
+        self.coreDataManager = coreDataManager
+        coreDataManager.managedObjectContext.persistentStoreCoordinator = coreDataManager.persistentStoreCoordinator
```

Comment:
This shouldn't be done in here. This specific initialization probably deserves a new CoreDataManager initializer?